src/op.cpp

```diff
@@ -452,44 +452,27 @@ uint32_t jitc_var_new_op(JitOp op, uint32_t n_dep, const uint32_t *dep) {
     bool dirty = false, literal = true, uninitialized = false, placeholder = false;
     uint32_t vti = 0;
     bool literal_zero[4] { }, literal_one[4] { };
-    uint32_t backend_i = 0;
     Variable *v[4] { };
     bool const_prop = jitc_flags() & (uint32_t) JitFlag::ConstProp;
 
     if (unlikely(n_dep == 0 || n_dep > 4))
         jitc_fail("jit_var_new_op(): 1-4 dependent variables supported!");
 
     for (uint32_t i = 0; i < n_dep; ++i) {
-        if (likely(dep[i])) {
+        uint32_t idx = dep[i];
+        if (likely(idx)) {
             Variable *vi = jitc_var(dep[i]);
             vti = std::max(vti, vi->type);
             size = std::max(size, vi->size);
             dirty |= (bool) vi->ref_count_se;
             placeholder |= (bool) vi->placeholder;
-            backend_i |= (uint32_t) vi->backend;
             v[i] = vi;
-
```

Comment:
Curious about this removal.

Reply:
This was moved down as we first need to check whether to use constant propagation or not.